### DISCUSSION TITLE
NetCDF to CSV for a polygon / AOI

### DISCUSSION THREAD

**Discussion by aswinkokkat** [opengeos/leafmap#314] [Q&A] [answered]
I am having a river basin as a study area with 22 sub-basins.  So I need to extract monthly rainfall data for each pixel falling in the 22 sub-basins. So as a user if i can  upload my area of interest(AOI) boundary file (shp file, a river basin with sub-basins) so the code can identify the pixels and extract the data in .csv for our study region. 

If the code does some optional work like zonal statistics it will be good.

**Comment by giswqs** [comment, marked as answer]
Use rasterio to read the image as an numpy array and convert it to a CSV. 

https://www.geeksforgeeks.org/convert-a-numpy-array-into-a-csv-file/
